File: app/src/main/java/com/example/schooldiary/ui/adapters/holidays/HolidaysHolder.java
package com.example.schooldiary.ui.adapters.holidays;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.schooldiary.databinding.ItemHolidaysBinding;
import com.example.schooldiary.model.Dates;

public class HolidaysHolder extends RecyclerView.ViewHolder {
    private View itemView;

    public HolidaysHolder(View itemView) {
        super(itemView);
        this.itemView = itemView;
    }

    public void bindData(Dates dates) {

        String[] date = dates.getTime().split(" ");

        ItemHolidaysBinding binding = ItemHolidaysBinding.bind(itemView);

        TextView name = binding.holidaysName;
        TextView start = binding.holidaysStart;
        TextView end = binding.holidaysEnd;

        name.setText(dates.getName());
        start.setText(date[0]);
        end.setText(date[2]);
    }
}

File: app/src/main/java/com/example/schooldiary/ui/adapters/holidays/HolidaysHolder.kt
package com.example.schooldiary.ui.adapters.holidays

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.schooldiary.databinding.ItemHolidaysBinding
import com.example.schooldiary.model.dates.Dates

class HolidaysHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val possibleDates: Set<String> = setOf("Winter", "Spring", "Summer", "Autumn")

    fun bindData(dates: Dates) {
        val date = dates.time.split(" ")
        val binding = ItemHolidaysBinding.bind(itemView)
        if (dates.name in possibleDates) {
            binding.name.text = dates.name
        }
        if (date.size <= 3) {
            binding.holidaysStart.text = date[0]
            binding.holidaysEnd.text = date[2]
        }
    }
}